bitcoinspv/clients/sui/sui_spv_client.go

```diff
@@ -92,167 +126,141 @@ func (c *SPVClient) InsertHeaders(ctx context.Context, blockHeaders []wire.Block
-func (c *SPVClient) ContainsBlock(ctx context.Context, blockHash chainhash.Hash) (bool, error) {
-	arguments := []interface{}{
-		c.lcObjectID,
-		BlockHashToHex(blockHash),
-	}
+func (c *BTCLightClientObject) ContainsBlock(ctx context.Context, blockHash chainhash.Hash) (bool, error) {
+	ptb := suiptb.NewTransactionDataTransactionBuilder()
 
-	response, err := c.moveCall(ctx, containsBlockFunc, arguments)
+	b, err := bcs.Marshal(blockHash[:])
 	if err != nil {
 		return false, err
 	}
 
-	eventData, err := c.extractFirstEvent(ctx, response.Effects.TransactionDigest)
+	err = ptb.MoveCall(
+		c.PackageID,
+		lcModule,
+		containsBlockFunc,
+		[]sui.TypeTag{},
+		[]suiptb.CallArg{c.LcObjArg, {Pure: &b}},
+	)
 	if err != nil {
 		return false, err
 	}
-	exist, ok := eventData["exist"].(bool)
-	if !ok {
-		return false, fmt.Errorf("unexpected event data format: 'exist' field not found or not a boolean")
-	}
 
-	return exist, nil
-}
+	resp, err := c.devInspectTransactionBlock(ctx, ptb)
 
-// GetLatestBlockInfo returns the block hash and height of the best block header.
-func (c *SPVClient) GetLatestBlockInfo(ctx context.Context) (*clients.BlockInfo, error) {
-	arguments := []interface{}{
-		c.lcObjectID,
+	if err != nil {
+		return false, err
 	}
 
-	response, err := c.moveCall(ctx, getChainTipFunc, arguments)
-	if err != nil {
-		return nil, err
+	if resp.Error != "" {
+		return false, errors.New(resp.Error)
 	}
 
-	eventData, err := c.extractFirstEvent(ctx, response.Effects.TransactionDigest)
+	resultEncoded := getBCSResult(resp)
 	if err != nil {
-		return nil, err
+		return false, err
 	}
 
-	lightBlockHashData, ok := eventData["light_block_hash"].([]interface{})
-	if !ok {
-		return nil, ErrLightBlockHashNotFound
-	}
+	var result bool
 
-	heightData, ok := eventData["height"]
-	if !ok {
-		return nil, ErrHeightNotFound
+	err = bcs.UnmarshalAll(resultEncoded[0], &result)
+	if err != nil {
+		return false, err
 	}
 
-	heightStr, ok := heightData.(string)
-	if !ok {
-		return nil, fmt.Errorf("%w: got %T", ErrHeightInvalidType, heightData)
-	}
+	return result, nil
+}
 
-	height, err := strconv.ParseInt(heightStr, 10, 64)
+// GetLatestBlockInfo returns the block hash and height of the best block header.
+func (c *BTCLightClientObject) GetLatestBlockInfo(ctx context.Context) (*clients.BlockInfo, error) {
+	ptb := suiptb.NewTransactionDataTransactionBuilder()
+
+	err := ptb.MoveCall(
+		c.PackageID,
+		lcModule,
+		headFunc,
+		[]sui.TypeTag{},
+		[]suiptb.CallArg{c.LcObjArg},
+	)
 	if err != nil {
-		return nil, fmt.Errorf("%w: %s", ErrHeightInvalidValue, err.Error())
+		return nil, err
 	}
 
-	hashBytes, err := extractHashBytes(lightBlockHashData)
+	resp, err := c.devInspectTransactionBlock(ctx, ptb)
 	if err != nil {
 		return nil, err
 	}
-
-	blockHash, err := chainhash.NewHash(hashBytes)
-	if err != nil {
-		return nil, fmt.Errorf("%w: %w", ErrBlockHashInvalid, err)
+	if !resp.Effects.Data.IsSuccess() {
```

Comment:
why don't checking `resp.Error`? Is it the same?

Reply:
Error is a string. IsSuccess() help us check status of transaction.